File: frontend/src/types/stock.ts
// ================================================================
// Stock-related TypeScript interfaces
// ================================================================

export interface StockInfo {
  ts_code: string
  name: string
  symbol: string
  industry?: string
  market?: string
  list_date?: string
  total_cap?: number
}

export interface KlineData {
  trade_date: string
  open: number
  high: number
  low: number
  close: number
  volume: number
  amount: number
  pct_change?: number
  turnover_rate?: number
}

export interface QuoteData {
  ts_code: string
  trade_date: string
  open: number
  high: number
  low: number
  close: number
  pre_close: number
  volume: number
  amount: number
  pct_change: number
  turnover_rate?: number
  close_raw?: number
  open_raw?: number
  high_raw?: number
  low_raw?: number
  pre_close_raw?: number
  price?: number
  open_price?: number
  high_price?: number
  low_price?: number
  pre_close_price?: number
}

export interface IndicatorData {
  ts_code: string
  trade_date: string
  macd_dif?: number
  macd_dea?: number
  macd_bar?: number
  rsi_6?: number
  rsi_12?: number
  rsi_24?: number
  kdj_k?: number
  kdj_d?: number
  kdj_j?: number
  ma5?: number
  ma10?: number
  ma20?: number
  ma60?: number
}

export interface StockSearchResult {
  value: string
  name: string
  ts_code: string
  industry?: string
}

File: backend/app/models/stock.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class StockBase(BaseModel):
    ts_code: str
    symbol: str
    name: str
    industry: Optional[str] = None
    market: Optional[str] = None
    list_date: Optional[str] = None


class StockCreate(StockBase):
    pass


class StockResponse(StockBase):
    class Config:
        from_attributes = True


class DailyQuote(BaseModel):
    ts_code: str
    trade_date: str
    open: float
    high: float
    low: float
    close: float
    volume: int
    amount: float
    turnover_rate: Optional[float] = None
    pct_change: Optional[float] = None
    pre_close: Optional[float] = None
    is_trading: bool = True
    adjust_flag: Optional[str] = "none"


class Indicator(BaseModel):
    ts_code: str
    trade_date: str
    ma_5: Optional[float] = None
    ma_10: Optional[float] = None
    ma_20: Optional[float] = None
    ma_60: Optional[float] = None
    macd_dif: Optional[float] = None
    macd_dea: Optional[float] = None
    macd_bar: Optional[float] = None
    rsi_6: Optional[float] = None
    rsi_12: Optional[float] = None
    rsi_24: Optional[float] = None
    kdj_k: Optional[float] = None
    kdj_d: Optional[float] = None
    kdj_j: Optional[float] = None
    boll_upper: Optional[float] = None
    boll_mid: Optional[float] = None
    boll_lower: Optional[float] = None
